pkg/apis/provisioning/v1alpha5/constraints.go

```diff
@@ -51,22 +51,48 @@ func (c *Constraints) ValidatePod(pod *v1.Pod) error {
 	if err := c.Taints.Tolerates(pod); err != nil {
 		return err
 	}
-	// Test if pod requirements are valid
 	requirements := NewPodRequirements(pod)
+	// Test if pod requirements are valid
 	if err := requirements.Validate(); err != nil {
 		return fmt.Errorf("invalid requirements, %w", err)
 	}
-	// Test if pod requirements are compatible
+	// Test if pod requirements are compatible to the provisioner
 	if errs := c.Requirements.Compatible(requirements); errs != nil {
 		return fmt.Errorf("incompatible requirements, %w", errs)
 	}
 	return nil
 }
 
+func (c *Constraints) GenerateLabels() *Constraints {
```

Comment:
Rather than mutating constraints, can we just return the map and allow the caller to use them as useful? This is approach is more stateless/functional.
```suggestion
func (c *Constraints) GenerateLabels() map[string]string {
```

Reply:
Agree. Done